components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt

```diff
@@ -324,6 +325,44 @@ class SystemEngineView @JvmOverloads constructor(
             session?.internalNotifyObservers { onContentPermissionRequest(SystemPermissionRequest(request)) }
         }
 
+        override fun onJsAlert(view: WebView?, url: String?, message: String?, result: JsResult): Boolean {
```

Comment:
Can all those parameters really be null?

Reply:
Unfortunately, this Apis are not annotated for null-availability and the [docs](https://developer.android.com/reference/android/webkit/WebChromeClient.html#onJsAlert(android.webkit.WebView,%20java.lang.String,%20java.lang.String,%20android.webkit.JsResult)) don't tell much neither :(

Reply:
@Amejia481 I [looked into this before](https://github.com/mozilla-mobile/android-components/pull/1466#discussion_r236026122) for nullability; the WebView is non-null for sure.

Reply:
Thanks @jonalmeida !